specutils/core/generic.py

```diff
@@ -0,0 +1,119 @@
+        """
+        self_kwargs = {"data": self._data, "dispersion": self._dispersion,
+                       "unit": self.unit, "wcs": self.wcs,
+                       "uncertainty": self._uncertainty, "mask": self.mask,
+                       "meta": self.meta}
+
+        self_kwargs.update(kwargs)
+
+        return self.__class__(**self_kwargs, copy=copy)
+
+    @classmethod
+    def from_array(cls, data, *args, **kwargs):
+        return cls(name="Array Data", data=data, **args, **kwargs)
+
+    @property
+    def dispersion(self):
+        """Dispersion values."""
+        if self._dispersion is None:
+            self._dispersion = np.arange(self.data.size)
+
+            if isinstance(self.wcs, WCS):
+                try:
+                    crval = self.wcs.wcs.crval[0]
+
+                    # RuntimeWarning: cdelt will be ignored since cd is present
+                    try:
+                        cdelt = self.wcs.wcs.cd[0][0]
+                    except AttributeError:
+                        cdelt = self.wcs.wcs.cdelt[0]
+
+                    end = self.data.shape[0] * cdelt + crval
+                    num = (end - crval) / cdelt
+
+                    # TODO: the values for the keywords are not guaranteed to
+                    # be at the first index
+                    if hasattr(self.wcs.wcs, 'ctype') and "log" \
+                            in self.wcs.wcs.ctype[-1].lower():
+                        self._dispersion = np.logspace(crval, end, num)
+                    else:
+                        self._dispersion = np.arange(crval, end, cdelt)
+                except AttributeError:
+                    logging.warning("Invalid FITS headers; constructing "
+                                    "default dispersion array.")
+
+        return self._dispersion
+
+    @property
+    def dispersion_unit(self):
+        """Unit of dispersion."""
+        if self._dispersion_unit is None:
+            try:
+                self._dispersion_unit = self.wcs.wcs.cunit[0]
+            except AttributeError:
+                logging.warning("No dispersion unit information in WCS.")
+
+                try:
+                    self._dispersion_unit = u.Unit(
+                        self.meta['header']['cunit'][0])
+                except KeyError:
+                    logging.warning("No dispersion unit information in meta.")
+
+                    self._dispersion_unit = u.Unit("")
+
+        return self._dispersion_unit
+
+    @dispersion_unit.setter
+    def dispersion_unit(self, value):
+        self._dispersion_unit = value
+
+    def __add__(self, other):
+        return self.add(other, handle_meta='first_found')
+
+    def __sub__(self, other):
+        return self.subtract(other, handle_meta='first_found')
+
+    def __mul__(self, other):
+        return self.multiply(other, handle_meta='first_found')
+
+    def __truediv__(self, other):
+        return self.divide(other, handle_meta='first_found')
```

Comment:
[IIRC](http://www.acronymfinder.com/Slang/IIRC.html) this was not implement in the `NDArithmeticMixin` to avoid forcing objects to interact like numpy arrays. When I tried adding these in ccdproc a couple years ago I ran into problems with trying to for the returned object to have the proper class if I, say, added a `Quantity` to a `CCDData` object. 

In `NDArithmeticMixin` the situation would be messier because you don't know a priori what else the objects might be combined with in operations or which type should get priority.

In some cases I could imagine these methods being very handy, but think it makes sense to implement them in derived classes like`GenericSpectrum1D` rather than at a higher level.